modules/service/src/main/scala/lucuma/odb/graphql/OdbMapping.scala

```diff
@@ -152,14 +163,54 @@ object OdbMapping {
             pool.map(AsterismService.fromSessionAndUser(_, user))
 
           override val observationService: Resource[F, ObservationService[F]] =
-            pool.map(ObservationService.fromSessionAndUser(_, user))
+            pool.map { s =>
+              val oms = ObservingModeServices.fromSession(s)
+              ObservationService.fromSessionAndUser(s, user, oms)
+            }
 
           override val programService: Resource[F, ProgramService[F]] =
             pool.map(ProgramService.fromSessionAndUser(_, user))
 
           override val targetService: Resource[F, TargetService[F]] =
             pool.map(TargetService.fromSession(_, user))
 
+          override val itcClientService: Resource[F, ItcClientService[F]] =
+            pool.map { s =>
+              val oms = ObservingModeServices.fromSession(s)
+              ItcClientService.fromSession(s, user, oms)
+            }
+
+          override def itcQuery(
+            path:     Path,
+            pid:      Program.Id,
+            oid:      Observation.Id,
+            useCache: Boolean
+          ): F[Json] = {
+
+            def toJson(res: List[ItcResult]): F[Json] =
+              res.partitionMap {
+                case ItcResult.Error(msg)           => msg.asLeft[ItcResult.Success]
+                case s @ ItcResult.Success(_, _, _) => s.asRight[String]
+              }.bimap(_.headOption, _.maxByOption(_.exposureTime.value)) match {
+                case (_, Some(success)) => success.asJson.pure[F]
+                case (Some(error), _)   => new RuntimeException(s"Could not obtain ITC results: $error").raiseError[F, Json]
+                case _                  => Json.Null.pure[F]
+              }
```

Comment:
Here we're taking a decision about how to handle multiple ITC results (picking the longest exposure time successful one) for an observation when there is more than one target in its asterism.  I think it might be good to offer results per target instead.